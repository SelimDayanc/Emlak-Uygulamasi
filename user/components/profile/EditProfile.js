/*
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {Image, StyleSheet, Text, TouchableHighlight, View} from 'react-native';

import colors from '../../../common/colors';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

export default class EditProfile extends Component {
  static propTypes = {
    loadScene: PropTypes.func.isRequired,
    user: PropTypes.oneOfType([PropTypes.object, PropTypes.null]).isRequired,
  };

  render() {
    const {loadScene, user} = this.props;
    return (
      <TouchableHighlight
        onPress={() => loadScene('user')}
        underlayColor="transparent">
        <View style={styles.container}>
          <View style={styles.leftCol}>
            <Text style={styles.username}>{user.name}</Text>
            <Text style={styles.hint}>View and edit profile</Text>
          </View>

          <View style={styles.rightCol}>
            {user.image
              ? <Image
                  source={{uri: user.image}}
                  style={styles.logo}
                  resizeMode="cover"
                />
              : <FontAwesome
                  name="user-circle-o"
                  color={colors.smokeGreyDark}
                  size={80}
                />}

          </View>
        </View>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 50,
  },
  leftCol: {
    flex: 2,
    alignItems: 'center',
    justifyContent: 'center',
  },
  rightCol: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  username: {
    fontSize: 20,
    color: colors.smokeGreyDark,
    fontWeight: '700',
    paddingVertical: 5,
  },
  hint: {
    color: colors.grey,
    fontWeight: '100',
    paddingBottom: 5,
  },
  icon: {
    fontWeight: '100',
  },
  logo: {
    height: 80,
    width: 80,
    borderRadius: 40,
  },
});
