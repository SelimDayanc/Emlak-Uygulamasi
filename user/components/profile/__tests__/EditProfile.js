import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EditProfile from './../EditProfile';

test('renders heart', () => {
  const tree = renderer
    .create(
      <EditProfile
        loadScene={() => {}}
        user={{name: 'name', email: 'z4ls@live.com'}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
