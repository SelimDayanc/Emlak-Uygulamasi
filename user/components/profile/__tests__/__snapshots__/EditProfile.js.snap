// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 50,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#5d5f63",
            "fontSize": 20,
            "fontWeight": "700",
            "paddingVertical": 5,
          }
        }
      >
        name
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#999",
            "fontWeight": "100",
            "paddingBottom": 5,
          }
        }
      >
        View and edit profile
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#5d5f63",
              "fontSize": 80,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        ïŠ¾
      </Text>
    </View>
  </View>
</View>
`;
